File: Add Code Here/LeetCode_Sols/Easy/104. Maximum Depth of Binary Tree/Recursion - 1/Solution.java
class Solution {
    public int maxDepth(TreeNode root) {
        return traverse(root, 0);
    }

    public int traverse(TreeNode node, int depth) {
        if(node == null) {
            return depth;
        }
        return Math.max(
            traverse(node.left, depth),
            traverse(node.right, depth)
        ) + 1;
    }
}

File: Add Code Here/LeetCode_Sols/Easy/104. Maximum Depth of Binary Tree/Recursion - 1/Solution.cpp
class Solution {
public:
    int maxDepth(TreeNode* root) {
        return traverse(root, 0);
    }

    int traverse(TreeNode* node, int depth) {
        if(!node) {
            return depth;
        }
        return max(
            traverse(node->left,depth),
            traverse(node->right,depth)
        ) + 1;
    }
};